cmd/docker/docker.go

```diff
@@ -72,17 +111,48 @@ func newDockerCommand(dockerCli *command.DockerCli) *cobra.Command {
 			ccmd.Println(err)
 		}
 	})
+}
 
-	flags = cmd.Flags()
-	flags.BoolVarP(&opts.Version, "version", "v", false, "Print version information and quit")
-	flags.StringVar(&opts.ConfigDir, "config", cliconfig.Dir(), "Location of client config files")
-	opts.Common.InstallFlags(flags)
+func setValidateArgs(dockerCli *command.DockerCli, cmd *cobra.Command, flags *pflag.FlagSet, opts *cliflags.ClientOptions) {
+	// The Args is handled by ValidateArgs in cobra, which does not allows a pre-hook.
+	// As a result, here we replace the existing Args validation func to a wrapper,
+	// where the wrapper will check to see if the feature is supported or not.
+	// The Args validation error will only be returned if the feature is supported.
+	visitAll(cmd, func(ccmd *cobra.Command) {
+		// if there is no tags for a command or any of its parent,
+		// there is no need to wrap the Args validation.
+		if !hasTags(ccmd) {
+			return
+		}
 
-	cmd.SetOutput(dockerCli.Out())
-	cmd.AddCommand(newDaemonCommand())
-	commands.AddCommands(cmd, dockerCli)
+		if ccmd.Args == nil {
+			return
+		}
```

Comment:
@dnephin The indentation seems to be OK. The issue is that diff is not able to find the logic `diff` piece:
```udiff
                        return
@@ -72,17 +100,52 @@ func newDockerCommand(dockerCli *command.DockerCli) *cobra.Command {
                        ccmd.Println(err)
                }
        })
+}
 
-       flags = cmd.Flags()
-       flags.BoolVarP(&opts.Version, "version", "v", false, "Print version information and quit")
-       flags.StringVar(&opts.ConfigDir, "config", cliconfig.Dir(), "Location of client config files")
-       opts.Common.InstallFlags(flags)
+func setValidateArgs(dockerCli *command.DockerCli, cmd *cobra.Command, flags *pflag.FlagSet, opts *cliflags.ClientOptions) {
+       // The Args is handled by ValidateArgs in cobra, which does not allows a pre-hook.
+       // As a result, here we replace the existing Args validation func to a wrapper,
+       // where the wrapper will check to see if the feature is supported or not.
+       // The Args validation error will only be returned if the feature is supported.
+       visitAll(cmd, func(ccmd *cobra.Command) {
+               // if there is no tags for a command or any of its parent,
+               // there is no need to wrap the Args validation.
+               if !hasTags(ccmd) {
+                       return
+               }
 
-       cmd.SetOutput(dockerCli.Out())
-       cmd.AddCommand(newDaemonCommand())
-       commands.AddCommands(cmd, dockerCli)
+               if ccmd.Args == nil {
+                       return
+               }
 
-       return cmd
+               cmdArgs := ccmd.Args
```